bluetooth/api/blip_api.py

```diff
@@ -177,70 +187,100 @@ def update_configs(all_analyses, dbset):
+                                               EXCLUDED.report_id,
+                                               EXCLUDED.report_name,
+                                               EXCLUDED.route_id ,
+                                               EXCLUDED.route_name ,
+                                               EXCLUDED.route_points); 
+                        '''
+            with conn:
+                with conn.cursor() as cur:  
+                    cur.execute(upsert_sql, row)
+      
+            analyses_pull_data[row['analysis_id']] = {'pull_data': True,
+                                                           'report_name': row['report_name']}
         except IntegrityError as err:
             LOGGER.error(err)
 
-    db.close()
-
+    conn.close()
     analyses_to_pull = {analysis_id: analysis for (
-        analysis_id, analysis) in analyses_pull_data.items() if analysis['pull_data']}
+                                                analysis_id, analysis) in analyses_pull_data.items() if analysis['pull_data']}
+
     return analyses_to_pull
 
 def move_data(dbset):
     try:
         try:
-            db = _get_db(dbset)
+            conn = _get_db(dbset)
         except RetryError as retry_err:
             LOGGER.critical('Number of retries exceeded to connect to DB with the error:')
             retry_err.reraise()
-        db.begin()
-        query = db.query("SELECT bluetooth.move_raw_data();")
-        if query.getresult()[0][0] != 1:
-            db.rollback()
-            raise DatabaseError('bluetooth.move_raw_data did not complete successfully')
-        query = db.query("SELECT king_pilot.load_bt_data();")
-        if query.getresult()[0][0] != 1:
-            db.rollback()
-            raise DatabaseError('king_pilot.load_bt_data did not complete successfully')
-        query = db.query("TRUNCATE bluetooth.raw_data;")
-        db.query('DELETE FROM king_pilot.daily_raw_bt WHERE measured_timestamp < now()::DATE;')
-        db.commit()
+        ## BT move raw data
+        with conn:
+            with conn.cursor() as cur:
+                cur.execute("SELECT bluetooth.move_raw_data();")
+                query = cur.fetchone()
+        if query[0] != 1:
+            conn.rollback()
+            raise DatabaseError('bluetooth.move_raw_data did not complete successfully') 
+        ## King pilot 
+        with conn:
+            with conn.cursor() as cur:
+                cur.execute("SELECT king_pilot.load_bt_data();")
+                query = cur.fetchone()    
+        if query[0] != 1:
+            conn.rollback()
+            raise DatabaseError('king_pilot.load_bt_data did not complete successfully') 
+        ## Truncate and delete if successful 
+        with conn:
+            with conn.cursor() as cur:
+                cur.execute("TRUNCATE bluetooth.raw_data;")
+                cur.execute("DELETE FROM king_pilot.daily_raw_bt WHERE measured_timestamp < now()::DATE;")
```

Comment:
is this `now()::DATE` bad? 👁️ 👁️

Reply:
zomg yes.
Also rather tempted to say we should delete all the `king_pilot` stuff, but mayyyyybe pulling the live data hourly can be useful again.